docs/application/dotnet/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1084 @@
+
+            /// Set
+            VoiceControlManagerClient.RecognitionMode = recognitionModeType;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+-   Get or set private data between voice control manager and voice control engine.
+    You can use the `GetPrivateData` and `SetPrivateData` Methods of the [Tizen.Uix.VoiceControlManager.VoiceControlManagerClient](https://samsung.github.io/TizenFX/latest/api/Tizen.Uix.VoiceControlManager.VoiceControlManagerClient.html) class.
+    This option must be set when voice control manager is in the `Ready` state.
+    ```
+    void GetSetPrivateData()
+    {
+        try
+        {
+            /// Get
+            string privateData = VoiceControlManagerClient.GetPrivateData("privateKey");
+
+            /// Set
+            VoiceControlManagerClient.SetPrivateData("privateKey", "privateData");
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+
+
+<a name="commands"></a>
+## Manage Commands
+
+To create a command group and commands:
+
+1.  Create a command group with a command group handle.
+    The command group can have multiple commands. Each command has a text and a type.
+    The group can have `Background`, `Exclusive`, `Foreground`, `System`, `SystemBackground`, and `Widget` type commands. For more information on the command types, see [Tizen.Uix.VoiceControlManager.CommandType](https://samsung.github.io/TizenFX/latest/api/Tizen.Uix.VoiceControlManager.CommandType.html) enumeration.
+
+    > **Note**   
+    >
+    > It has command priority in order of `system`, `widget`, `foreground`, `system_background` and `background`.
+    Normally, The `system` and `system_background` commands is only registered in voice control manager.
+    The `foreground` and `background` commands are used in the vc client app.
+    The `exclusive` coomands is used starting with `VoiceControlManagerClient.Start(true)`.
+    The `widget` command is automatically registered in the elementary on the screen.
+
+    You can create the command group:
+
+    ```
+    void CreateCommandGroup()
+    {
+        try
+        {
+            VoiceCommandsGroup group = new VoiceCommandsGroup();
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+2.  Create a command.
+
+    First create a command handle, and then define the command and type:
+
+    ```
+    void CreateCommand()
+    {
+        try
+        {
+            VoiceCommand command = new VoiceCommand();
+
+            command.Command = "open";
+            command.Type = CommandType.Foreground;
+        }
+        catch (Exception e)
+        {
+            /// Error handling
+        }
+    }
+    ```
+
+3.  Add the command to the command group.
+
+    If necessary, you can also remove commands from the command group:
```

Comment:
To make it more clear can we put this line above the `RemoveCommand()` code?
For example:

3. Add the command to the command group:

    ```
    void AddCommand()
    {
        try
        {
            VoiceCommandsGroup group = new VoiceCommandsGroup();
            VoiceCommand command = new VoiceCommand();
            command.Command = "open";
            command.Type = CommandType.Foreground;
            group.Commands.Add(command);
        }
        catch (Exception e)
        {
            /// Error handling
        }
    }
    ```

    If necessary, you can also remove commands from the command group:

    ```
    void RemoveCommand()
    {
        try
        {
            VoiceCommand command = new VoiceCommand();
            command.Command = "open";
            command.Type = CommandType.Foreground;
            /// Other actions
            group.Commands.Remove(command);
        }
        catch (Exception e)
        {
            /// Error handling
        }
    }
     ```

Please verify and share your opinion.